web3/providers/ipc.py

```diff
@@ -245,7 +245,6 @@ def make_request(self, method: RPCEndpoint, params: Any) -> RPCResponse:
                     try:
                         raw_response += sock.recv(4096)
                     except socket.timeout:
-                        timeout.sleep(0)
```

Comment:
Without this sleep, the the `with Timeout(...) as timeout` block won't get a chance to raise the actual timeout.  This could result in an infinite loop if the socket continually times out on the call to `socket.recv(...)` as the context manager would never get a chance to enforce the timeout.

I don't think removing this timeout is appropriate.